docs/source/frameworks/xgboost.rst

```diff
@@ -2,49 +2,174 @@
+
+   # create a runner from the saved Booster
+   runner = bentoml.xgboost.get("booster_tree:latest").to_runner()
+
+   # create a BentoML service
+   svc = bentoml.Service("cancer_classifier", runners=[runner])
+
+   # define a new endpoint on the BentoML service
+   @svc.api(input=NumpyNdarray(), output=NumpyNdarray())
+   def classify_tumor(input: np.ndarray) -> np.ndarray:
+       # use 'runner.predict.run(input)' instead of 'booster.predict'
+       res = runner.predict.run(input)
+       return res
+
+Take note of the name of the service (``svc`` in this example) and the name of the file.
+
+You should also have a ``bentofile.yaml`` alongside the service file that specifies that
+information, as well as the fact that it depends on XGBoost. This can be done using either
+``python`` (if using pip), or ``conda``:
 
-   You can find more examples for **XGBoost** in our `gallery <https://github.com/bentoml/gallery>`_ repo.
+.. tab-set::
+   .. tab-item:: pip
 
-.. currentmodule:: bentoml.xgboost
+      .. code-block:: yaml
 
-.. autofunction:: bentoml.xgboost.save_model
+         service: "service:svc"
+         description: "My XGBoost service"
+         python:
+         - xgboost
 
-.. autofunction:: bentoml.xgboost.load_model
+   .. tab-item:: conda
+
+      .. code-block:: yaml
+
+         service: "service:svc"
+         description: "My XGBoost service"
+         conda:
+           channels:
+           - conda-forge
+           dependencies:
+           - xgboost
+
+Using Runners
+~~~~~~~~~~~~~
+.. seealso::
+
+   :ref:`concepts/runner:Using Runners`: a general introduction to the Runner concept and its usage.
+
+A runner for a Booster is created like so:
+
+.. code-block:: python
+
+   bentoml.xgboost.get("model_name:model_version").to_runner()
+
+``runner.predict.run`` is generally a drop-in replacement for ``booster.predict``. However, while it
+is possible to pass a ``DMatrix`` as input, BentoML does not support adaptive batching in that case.
+It is therefore recommended to use a NumPy ``ndarray`` or Pandas ``DataFrame`` as input instead.
+
+There are no special options for loading XGBoost.
+
+Runners must to be initialized in order for their ``run`` methods to work. This is done by BentoML
+internally when you serve a bento with ``bentoml serve``. See the :ref:`runner debugging guide
+<concepts/service:Debugging Runners>` for more information about initializing runners locally.
+
+
+GPU Inference
+~~~~~~~~~~~~~
+
+If there is a GPU available, the XGBoost Runner will automatically use ``gpu_predictor`` by default.
+This can be disabled by using the
+:ref:`BentoML configuration file <guides/configuration:Configuring BentoML>` to disable Runner GPU
+access:
+
+.. code-block:: yaml
+
+   runners:
+      # resources can be configured at the top level
+      resources:
+         nvidia.com/gpu: 0
+      # or per runner
+      my_runner_name:
+         resources:
+             nvidia.com/gpu: 0
+
+
+Adaptive Batching
```

Comment:
Should we link to the adaptive batching docs in the following section? Otherwise looks good